### PR TITLE
[Example] Update mass_spring_3d_ggui.py to v2

### PR DESCRIPTION
**Pull request by AlexBrown42**
Made some minor improvements there:

1. Better collision handling
2. Bending springs
3. Added dashpot_damping
4. Reset animation when the cloth moves completely out of the window

@AmesingFlank could you please help review this? Thanks.

### CODE REVIEW

File: cmake/TaichiCore.cmake [hunk 1]
```diff
@@ -403,5 +403,4 @@ if(TI_WITH_GGUI)
     include_directories(SYSTEM ${IMGUI_DIR} ${IMGUI_DIR}/backends ..)
     add_library(imgui  ${IMGUI_DIR}/backends/imgui_impl_glfw.cpp ${IMGUI_DIR}/backends/imgui_impl_vulkan.cpp ${IMGUI_DIR}/imgui.cpp ${IMGUI_DIR}/imgui_draw.cpp  ${IMGUI_DIR}/imgui_tables.cpp ${IMGUI_DIR}/imgui_widgets.cpp)
     target_link_libraries(${CORE_LIBRARY_NAME} imgui)
-
```
Comment (AmesingFlank) [inline cmake/TaichiCore.cmake:406]: is this included accidentally? :D

**Comment by AmesingFlank** [comment]
Hi! Thanks for these improvements. The file `mass_spring_3d_ggui.py` is used as an accompanying code sample for a [blog article](https://docs.taichi.graphics/blog), so I think it's the best to not modify it. Instead, how about we add your improved program in a new file, perhaps `mass_spring_3d_ggui_improved.py`?

**Review by AmesingFlank — APPROVED** [review]

**Review by bobcao3 — CHANGES_REQUESTED** [review]
Thanks for optimizing! Just need to remove the accidental cmake change and then rename this file

**Review by neozhaoliang — APPROVED** [review]
LGTM!

**Pull request merged**